File: modules/spring-core/testdata/java/service/beanRegistrarJava/com/app/MyBeanRegistrar.java
package com.app;

import org.springframework.beans.factory.BeanRegistrar;
import org.springframework.beans.factory.BeanRegistry;
import org.springframework.core.env.Environment;

public class MyBeanRegistrar implements BeanRegistrar {
    @Override
    public void register(BeanRegistry registry, Environment env) {
        registry.registerBean("foo", Foo.class);
        registry.registerBean(Bar.class);
        if (true) {
            registry.registerBean(Baz.class);
        }
    }
}

File: modules/spring-core/testdata/kotlin/service/beanRegistrarKotlin/com/app/MyBeanRegistrar.kt
package com.app

import org.springframework.beans.factory.dsl.BeanRegistrarDsl

class MyBeanRegistrar : BeanRegistrarDsl({
    registerBean<Foo>()
    registerBean<Bar>(name = "bar")
    profile("baz") {
        registerBean { Baz("Hello World!") }
    }
})
